src/components/TextInput/BaseTextInput/index.native.tsx

```diff
@@ -21,37 +21,71 @@ import useStyleUtils from '@styles/useStyleUtils';
 import useThemeStyles from '@styles/useThemeStyles';
 import variables from '@styles/variables';
 import CONST from '@src/CONST';
-import * as baseTextInputPropTypes from './baseTextInputPropTypes';
-
-function BaseTextInput(props) {
+import type BaseTextInputProps from './types';
+import type {BaseTextInputRef} from './types';
+
+function BaseTextInput(
+    {
```

Comment:
@fabioh8010 added for `maxLength` `undefined` because this is the type from RN. For `shouldUseDefaultValue` and `shouldSaveDraft` I spoke with @blazejkustra and we agreed that the default values are not needed for those props.